source/material_model/viscoelastic_plastic.cc

```diff
@@ -0,0 +1,796 @@
+                                     "compositional field is called stress_zz."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_xy") == 3,
+                          ExcMessage("Material model Viscoelastic Plastic only works if the fourth "
+                                     "compositional field is called stress_xy."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_xz") == 4,
+                          ExcMessage("Material model Viscoelastic Plastic only works if the fifth "
+                                     "compositional field is called stress_xz."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_yz") == 5,
+                          ExcMessage("Material model Viscoelastic Plastic only works if the sixth "
+                                     "compositional field is called stress_yz."));
+            }
+          else
+            ExcNotImplemented();
+
+          // It does not make sense to use this material with the 'no Advection, iterated Stokes'
+          // or 'single Advection, no Stokes' solver schemes.
+          AssertThrow((this->get_parameters().nonlinear_solver ==
+                       Parameters<dim>::NonlinearSolver::single_Advection_single_Stokes
+                       ||
+                       this->get_parameters().nonlinear_solver ==
+                       Parameters<dim>::NonlinearSolver::single_Advection_iterated_Stokes
+                       ||
+                       this->get_parameters().nonlinear_solver ==
+                       Parameters<dim>::NonlinearSolver::iterated_Advection_and_Stokes
+                       ||
+                       this->get_parameters().nonlinear_solver ==
+                       Parameters<dim>::NonlinearSolver::iterated_Advection_and_Newton_Stokes),
+                      ExcMessage("The material model will now work with the nonlinear "
+                                 "solver schemes 'no Advection, iterated Stokes' and "
+                                 "'single Advection, no Stokes'."));
+
+          // Functionality to average the additional RHS terms over the cell is not implemented.
+          // This enforces that the variable 'Material averaging' is set to 'none'.
+          AssertThrow((this->get_parameters().material_averaging ==
+                       MaterialModel::MaterialAveraging::none),
+                      ExcMessage("The viscoelastic_plastic material model cannot be used with "
+                                 "material averaging. The variable 'Material averaging' "
+                                 "in the 'Material model' subsection must be set to 'none'."));
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+
+
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = NonlinearDependence::compositional_fields;
```

Comment:
viscosity also depends on pressure and strain rate  (and in the future temperature)